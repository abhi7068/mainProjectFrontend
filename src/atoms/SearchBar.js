import React, { useState } from 'react';
import IconButton from '@material-ui/core/IconButton';
import { withRouter } from 'react-router-dom';
import SearchIcon from '@material-ui/icons/Search';
import { makeStyles } from '@material-ui/core';
import Async from 'react-select/async';
import Product from './Manipulations/product';
import { getproducts } from '../services/productService';
import { updateProducts } from '../store/actions/productAction';
import kebab from '../services/kebab';
import PropTypes from 'prop-types';

import debounce from 'lodash/debounce';
import { connect } from 'react-redux';

const useStyles = makeStyles((theme) => ({
  grow: {
    flexGrow: 1,
  },
  show: {
    width: '28vw',
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    display: 'flex',

    '&:hover': {
      // backgroundColor: fade(theme.palette.common.white, 2.5),
    },
    marginLeft: 0,
    marginRight: 0,
  },
  inputRoot: {
    color: 'inherit',
  },
  inputInput: {
    padding: theme.spacing(1, 1, 1, 0),
    paddingLeft: `calc(1em + ${theme.spacing(4)}px)`,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: '100px',
    },
  },
}));
function SearchBar(props) {
  const [searchvalue, setValue] = useState('');
  const handleSearch = () => {
    if (searchvalue) {
      getproducts(searchvalue)
        .then((res) => props.dispatch(updateProducts(res.data)))
        .then(props.history.push(`/search/${searchvalue}`), setValue(''));
    }
  };
  const loadProducts = debounce(
    (inputValue, callback) => {
      setValue(inputValue);
      getproducts(inputValue)
        .then((json) => {
          callback(json.data.map((product) => new Product(product)));
        })
        .catch((errors) => {});
    },
    1000,
    false
  );
  const handleSelection = (value) => {
    // let {onModelChange} = this.props;
    // onModelChange({target: {value: value, name: field}});

    if (value) {
      //  props.dispatch(productDetails(value._id));
      props.history.push(
        kebab(
          `/${
            value.category && value.category._id
              ? value.category
              : 'No Category'
          }/${value._id}`
        )
      );
      setValue('');
    }
  };
  const classes = useStyles();
  return (
    <div>
      <div className={classes.search}>
        {/* <div style={{ }}> */}
        <Async
          openMenuOnClick={false}
          menuPortalTarget={document.body}
          className={classes.show}
          components={{
            DropdownIndicator: () => null,
            IndicatorSeparator: () => null,
            NoOptionsMessage: () => 'No products ',
          }}
          loadOptions={loadProducts}
          isClearable
          value={Product}
          placeholder="Search"
          onChange={handleSelection}
        />
        {/* </div> */}
        <form onSubmit={handleSearch}>
          <IconButton aria-label="Search" onClick={handleSearch}>
            <SearchIcon
              style={{ color: 'white' }}
              className={classes.searchIcon}
            />
          </IconButton>
        </form>
      </div>
    </div>
  );
}
SearchBar.propTypes = {
  history: PropTypes.object,
  dispatch: PropTypes.func,
};
function mapStateToProps(state) {
  return {
    cart: state.cart,
    userDetails: state.userDetails,
  };
}
export default withRouter(connect(mapStateToProps)(SearchBar));
