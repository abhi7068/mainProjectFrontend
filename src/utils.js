import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
