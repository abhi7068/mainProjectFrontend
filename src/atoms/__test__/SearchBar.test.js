import { render } from '../../utils';

import SearchBar from '../SearchBar';

import React from 'react';

it('render', () => {
  render(<SearchBar />);
});

test('matches snapshot 2', () => {
  // axiosMock.get.mockImplementationOnce(() => Promise.resolve(response.data));

  const { asFragment } = render(<SearchBar />);
  expect(asFragment(<SearchBar />)).toMatchSnapshot();
});
