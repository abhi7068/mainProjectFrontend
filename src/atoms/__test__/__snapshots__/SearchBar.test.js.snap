// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 2 1`] = `
<DocumentFragment>
  <div>
    <div
      class="makeStyles-search-39"
    >
      <div
        class="makeStyles-show-38 css-2b097c-container"
      >
        <div
          class=" css-yk16xz-control"
        >
          <div
            class=" css-g1d714-ValueContainer"
          >
            <div
              class=" css-1wa3eu0-placeholder"
            >
              Search
            </div>
            <div
              class="css-b8ldur-Input"
            >
              <div
                class=""
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-3-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          />
        </div>
      </div>
      <form>
        <button
          aria-label="Search"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              style="color: white;"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
